src/Pandell.PackageRunner/Pandell.PackageRunner.csproj

```diff
@@ -0,0 +1,127 @@
+      <HintPath>..\..\nuget_modules\NuGet.Core.2.8.6\lib\net40-Client\NuGet.Core.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="System" />
+    <Reference Include="System.ComponentModel.Composition" />
+    <Reference Include="System.Core" />
+    <Reference Include="System.Web.Extensions" />
+    <Reference Include="WindowsBase" />
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="AssemblyChecker.cs" />
+    <Compile Include="Extensions.cs" />
+    <Compile Include="FileLogging.cs" />
+    <Compile Include="NugetCredentialProvider.cs" />
+    <Compile Include="Program.cs" />
+    <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="..\AssemblyVersionInfo.cs">
+      <Link>Properties\AssemblyVersionInfo.cs</Link>
+    </Compile>
+    <Compile Include="Token.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="packages.config">
+      <SubType>Designer</SubType>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <EmbeddedResource Include="..\..\nuget_modules\Microsoft.Web.Xdt.2.1.1\lib\net40\Microsoft.Web.XmlTransform.dll">
+      <Link>Properties\Microsoft.Web.XmlTransform.dll</Link>
+    </EmbeddedResource>
+    <EmbeddedResource Include="..\..\nuget_modules\NuGet.Core.2.8.6\lib\net40-Client\NuGet.Core.dll">
+      <Link>Properties\NuGet.Core.dll</Link>
+    </EmbeddedResource>
+  </ItemGroup>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <UsingTask TaskName="UpdateTokenFile" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v4.0.dll">
+    <ParameterGroup>
+      <Token ParameterType="System.String" Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System.IO"/>
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+            var tokenString = (this.Token ?? "").Replace("Public key token is ", "");
+            var tokenBytes = "";
+            for (var i = 0; i < tokenString.Length; i += 2)
+            {
+                tokenBytes += "0x" + tokenString.Substring(i, 2) + ", ";
+            }
+            var tokenFileLines = File.ReadAllLines("Token.cs");
+            if (tokenFileLines.Length != 10)
+            {
+                throw new InvalidOperationException("Token.cs file was not in the correct format (most likely missing or manually modified).");
+            }
+            if (!tokenFileLines[7].Contains(tokenBytes))
+            {
+                tokenFileLines[7] = "        public static readonly byte[] Bytes = { " + tokenBytes + "};"; 
+                File.WriteAllLines("Token.cs", tokenFileLines);
+                Log.LogMessage("Updating \"Token.cs\" to \"{0}\"", tokenString);
+            }
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+  <Target Name="UpdatePublicKeyToken" BeforeTargets="BeforeBuild">
+    <GetFrameworkSdkPath>
```

Comment:
Please use `Condition=" '$(WindowsSdkPath)' == '' "`. This will allow us to explicitly specify path to `sn.exe` in continuous integration builds.